astroid/inference.py

```diff
@@ -562,12 +562,48 @@ def _is_not_implemented(const):
     return isinstance(const, nodes.Const) and const.value is NotImplemented
 
 
+def _infer_old_style_string_formatting(
+    instance: nodes.Const, other: nodes.NodeNG, context: InferenceContext
```

Comment:
I thought InferenceContext was usually optional.

Reply:
Due to `callee` this is not the case. See L602 just before the `isinstance`. Such operations can't happen when `context` is `None`.